quic/s2n-quic-core/src/recovery/congestion_controller/fuzz_target.rs

```diff
@@ -0,0 +1,223 @@
+            let packet_info = self.subject.on_packet_sent(
+                self.timestamp,
+                bytes_sent as usize,
+                app_limited,
+                &self.rtt_estimator,
+            );
+            self.sent_packets.push_back(SentPacketInfo {
+                sent_bytes: bytes_sent,
+                time_sent: self.timestamp,
+                cc_packet_info: packet_info,
+            });
+        }
+    }
+
+    fn on_ack_received<Rnd: random::Generator>(&mut self, count: u8, rng: &mut Rnd) {
+        for _ in 0..count {
+            if let Some(sent_packet_info) = self.sent_packets.pop_front() {
+                self.subject.on_ack(
+                    sent_packet_info.time_sent,
+                    sent_packet_info.sent_bytes as usize,
+                    sent_packet_info.cc_packet_info,
+                    &self.rtt_estimator,
+                    rng,
+                    self.timestamp,
+                );
+            } else {
+                break;
+            }
+        }
+    }
+
+    fn on_packet_lost<Rnd: random::Generator>(&mut self, rng: &mut Rnd) {
+        if let Some(sent_packet_info) = self.sent_packets.pop_front() {
+            self.subject.on_packet_lost(
+                sent_packet_info.sent_bytes as u32,
+                sent_packet_info.cc_packet_info,
+                false,
+                false,
+                rng,
+                self.timestamp,
+            );
+        }
+    }
+
+    fn on_rtt_updated(&mut self, rtt: Duration) {
+        if let Some(sent_packet_info) = self.sent_packets.front() {
+            self.rtt_estimator.update_rtt(
+                Duration::ZERO,
+                rtt,
+                self.timestamp,
+                false,
+                PacketNumberSpace::Initial,
+            );
+            self.subject.on_rtt_update(
+                sent_packet_info.time_sent,
+                self.timestamp,
+                &self.rtt_estimator,
+            );
+        }
+    }
+
+    fn on_explicit_congestion(&mut self, ce_count: u64) {
+        let ce_count = ce_count.min(self.sent_packets.len() as u64);
+        self.subject
+            .on_explicit_congestion(ce_count, self.timestamp)
+    }
+
+    fn on_packet_discarded(&mut self) {
+        if let Some(sent_packet_info) = self.sent_packets.pop_front() {
+            self.subject
+                .on_packet_discarded(sent_packet_info.sent_bytes as usize)
+        }
+    }
+
+    fn on_mtu_updated(&mut self, mtu: u16) {
+        self.subject.on_mtu_update(mtu)
+    }
+
+    fn invariants(&self) {
+        let bytes_in_flight: u32 = self
+            .sent_packets
+            .iter()
+            .map(|sent_packet_info| sent_packet_info.sent_bytes as u32)
+            .sum();
+
+        assert_eq!(bytes_in_flight, self.subject.bytes_in_flight());
+    }
+}
+
+#[test]
+fn cubic_fuzz() {
+    check!()
+        .with_generator((MINIMUM_MTU..=9000, 0..255, gen::<Vec<Operation>>()))
```

Comment:
You should be able to do an unbounded seed here, yeah?

```suggestion
        .with_generator((MINIMUM_MTU..=9000, gen(), gen::<Vec<Operation>>()))
```

Reply:
Yep, I wasn't sure how to do that